tests/protocol/test_imds.py

```diff
@@ -20,18 +20,18 @@
 import os
 import unittest
 
-import azurelinuxagent.common.protocol.imds as imds
+from azurelinuxagent.common.protocol import imds
 
 from azurelinuxagent.common.datacontract import set_properties
 from azurelinuxagent.common.exception import HttpError, ResourceGoneError
 from azurelinuxagent.common.future import ustr, httpclient
 from azurelinuxagent.common.utils import restutil
-from tests.ga.test_update import ResponseMock
```

Comment:
There are multiple implementations of mock HTTP responses that make things confusing and cause some really weird test failures. Removing the usage of this one in favor of MockHttpResponse, which provides a more complete implementation